transcom/mymove
# [MB-16004] Add DOD ID number to EDI N9 reference header

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16004)

## Summary

This PR adds the service member's DOD ID number to the EDI N9 reference header

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

Please refer to the [Acceptance Testing doc](https://transcom.github.io/mymove-docs/docs/backend/testing/acceptance-testing-syncada-edi-invoicing) for EDI generation or more information re: `prime-api-client` commands

1. Go to the office app
2. Login as a new TIO with gbloc KKFA 
3. Search for move code "RDY4PY" and click in
4. Scroll down to payment requests, click Review, and approve one
5. In your terminal, run `prime-api-client --insecure create-payment-request --filename ./payload.json | jq '.id,.paymentRequestNumber'` which should give you the 9 digit payment request ID for the following command
6. Run `make bin/prime-api-client bin/generate-payment-request-edi && generate-payment-request-edi --payment-request-number 6682-5285-1` replacing the end numbers with the payment request ID from previous
7. Confirm that the EDI prints in your terminal and includes the new N9 field
8. Confirm `make server_test` runs without failure

### Frontend

There were no FE changes

### Backend


### Database

#### Any new migrations/schema changes:

There were no migration nor scheme changes

## Screenshots
![Screenshot 2023-05-18 at 11 35 31 AM](https://github.com/transcom/mymove/assets/16230705/93fba172-7e08-43fb-a16b-b5d5181e9f60)
